### ISSUE TITLE
train error

### ISSUE THREAD

**Issue by skyfallsss** [bhpfelix/segment-anything-finetuner#7]
lightning_lite.utilities.exceptions.MisconfigurationException: The provided lr scheduler `LambdaLR` doesn't follow PyTorch's LRScheduler API. You should override the `LightningModule.lr_scheduler_step` hook with your own logic if you are using a custom LR scheduler.

**Comment by aymanaboghonim** [comment]
did u solve it ???

**Comment by shivarajkarki** [comment]
Work around is install latest version of tranformers and pytorch lightening, in my case `tranformers==4.30.2` and `lightning==2.0.5`, in this case need to do some changes in code. in finetune.py @273 change `validation_epoch_end` to `on_validation_epoch_end`  and inside trainer change `strategy='ddp' if NUM_GPUS > 1 else None` to `strategy='ddp' if NUM_GPUS > 1 else 'auto'`

**Comment by Sencc** [comment]
@shivarajkarki  Thanks for your answer, it works for me

**Comment by skyfallsss** [comment]
what about the version of the pytorch-lightning?

**Comment by skyfallsss** [comment]
/data/home/cuitor/anaconda3/envs/sam/lib/python3.8/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:359: UserWarning: `ModelCheckpoint(monitor='val_per_mask_iou')` could not find the monitored key in the returned metrics: ['loss', 'loss_focal', 'loss_dice', 'loss_iou', 'train_per_mask_iou', 'epoch', 'step']. HINT: Did you call `log('val_per_mask_iou', value)` in the `LightningModule`?
My problem is caused by pytorch_lightning.

**Comment by Sencc** [comment]
@skyfallsss This is just a UserWarning, and it has no effect on the code running.

**Comment by skyfallsss** [comment]
When I use one GPU, It's going to be a segmentation error after few validation.

**Comment by skyfallsss** [comment]
Epoch 0:   0%| | 1/620 [00:07<1:13:26,  7.12s/it, v_num=27, loss=0.144, loss_focal=0.00114, 07/31/2023 20:44:04 - INFO - torch.nn.parallel.distributed -   Reducer buckets have been rebuilt in this iteration.
Epoch 0:   8%| | 50/620 [02:22<26:59,  2.84s/it, v_num=27, loss=0.191, loss_focal=0.0018, los/data/home/cuitor/anaconda3/envs/sam/lib/python3.8/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:359: UserWarning: `ModelCheckpoint(monitor='val_per_mask_iou')` could not find the monitored key in the returned metrics: ['loss', 'loss_focal', 'loss_dice', 'loss_iou', 'train_per_mask_iou', 'epoch', 'step']. HINT: Did you call `log('val_per_mask_iou', value)` in the `LightningModule`?
  warning_cache.warn(m)
                                                                                            Segmentation fault (core dumped)█████████████▉              | 137/266 [06:06<05:44,  2.67s/it]
